README.md

```diff
@@ -1,39 +1,98 @@
-This should run the program and print on your console.
+- **Month Menu:** A user-friendly menu where all expense tracking is facilitated. The month menu is the hub for 
+expense and category creation, viewing previous expense logs, and generating the graphical summary. 
 
-You will find HelloWorldTest in `src/test/java/tutorial` directory. Right click on the HelloWorldTest file and click on `Run HelloWorldTest`.
-All tests should pass. Your team can remove this sample of how testing works once you start adding your project code to the repo.
+- **Graphical Monthly Reports:** A comprehensive graphical representation of the user's spending distribution. 
+Generating easy-to-understand graphs through UI frameworks, users gain a visual insight into their financial allocation 
+across different categories.
 
-Moving forward, we expect you to maintain this project structure. You *should* use Gradle as the build environment, but it is fine if your team prefers to use something else -- just remove the gradle files and push your preferred project setup. Assuming you stick with Gradle, your source code should go into `src/main/java` (you can keep creating more subdirectories as per your project requirement). Every source class can auto-generate a test file for you. For example, open HelloWorld.java file and click on the `HelloWorld` variable as shown in the image below. You should see an option `Generate` and on clicking this your should see an option `Test`. Clicking on this will generate a JUnit test file for `HelloWorld` class. This was used to generate the `HelloWorldTest`.
+- **Persistent Data Storage:** This program ensures continuity by storing user data onto their local file system, 
+allowing access and maintenance across various sessions. This feature guarantees that the financial history and insights
+accumulated remain preserved and accessible over time.
 
-![image](https://user-images.githubusercontent.com/5333020/196066655-d3c97bf4-fdbd-46b0-b6ae-aeb8dbcf351d.png)
+## Installation
 
-You can create another simple class and try generating a test for this class.
+1. Clone the repository: `git clone https://github.com/CSC207-2023Y-UofT/course-project-DoorKeepers.git`
+2. Locate local file in your terminal or open in an IDE.
+3. Navigate to the correct directory: `cd src/main/java`
+4. Compile the project: `javac Main.java`
+5. Run the project: `java Main`
+
+## Abbreviations
+| Name              | Abbreviation |
+|-------------------|--------------|
+| Controller        | C            |
+| UseCaseInteractor | UCI          |
+| InputBoundary     | IB           |
+| InputData         | ID           |
+| OutputBoundary    | OB           |
+| OutputData        | OD           |
+| Presenter         | P            |
+| ViewBoundary      | VB           |
+| View              | V            |
+| Gateway           | G            |
+
+
+## Clean Architecture Specifications
+
+### Clean Architecture Violations
```

Comment:
Expense and Category are basic object types we have for conveying information that are not separable. For instance, a expense called "Walmart" would have no meaning unless we know the specific amount of money spent shopping in Walmart. 
These classes belong to the Entity layer, but are passed around in both the use_case and views packages, which is a Clean Architecture violation. We could remove the imports to these entities, and change the output data to Object[][] type in the UpdateViewUCI to accommodate Java Swing requirements. However, we thought it is not adhering to SOLID principle from the point that Object[][] may not be the type required by other view libraries that could be imported, which is in contrary to the Open/Closed Principle.

Reply:
Thanks!